src/constraint/encoding.ts

```diff
@@ -140,6 +140,34 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true;
     }
   },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: [Property.SCALE, Property.SCALE_TYPE, Property.TYPE],
+    allowEnumSpecForProperties: true,
+    strict: true,
+    satisfy: (encQ: EncodingQuery, schema: Schema, opt: QueryConfig) => {
+      if (encQ.scale) {
+        const scale: ScaleQuery = encQ.scale as ScaleQuery;
+        const hasScaleType: boolean = (scale.type !== undefined);
+
+        if ((!hasScaleType && isEnumSpec(encQ.type)) || isEnumSpec(scale.type)) {
+          return true;
+        }
+
+        for (let scaleProp in scale) {
+          if (SUPPORTED_SCALE_PROPERTY_INDEX[scaleProp]) {
```

Comment:
is there any scale property that's not in the `SUPPORTED_SCALE_PROPERTY_INDEX`?    

I can think of one case (guess what I can thought of) ?